### PR TITLE
Fix MC-124320

### PR DESCRIPTION
**Pull request by BillyGalbreath**
https://bugs.mojang.com/browse/MC-124320

### CODE REVIEW

File: Spigot-Server-Patches/0348-Fix-MC-124320.patch [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+diff --git a/src/main/java/net/minecraft/server/EntityEnderman.java b/src/main/java/net/minecraft/server/EntityEnderman.java
+index 2db0eb494..d2eb4dfdf 100644
+--- a/src/main/java/net/minecraft/server/EntityEnderman.java
++++ b/src/main/java/net/minecraft/server/EntityEnderman.java
+@@ -343,7 +343,7 @@ public class EntityEnderman extends EntityMonster {
+             BlockPosition blockposition = new BlockPosition(i, j, k);
+             IBlockData iblockdata = world.getType(blockposition);
+             IBlockData iblockdata1 = world.getType(blockposition.down());
+-            IBlockData iblockdata2 = this.a.getCarried();
++            IBlockData iblockdata2 = Block.b(this.a.getCarried(), this.a.world, blockposition);
```
Comment (stephan-gh) [inline Spigot-Server-Patches/0348-Fix-MC-124320.patch]: `// Paper - Fix MC-124320`

**Review by stephan-gh — CHANGES_REQUESTED** [review]

File: Spigot-Server-Patches/0348-Fix-MC-124320.patch [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+diff --git a/src/main/java/net/minecraft/server/EntityEnderman.java b/src/main/java/net/minecraft/server/EntityEnderman.java
+index 2db0eb494..a6cd85ff9 100644
+--- a/src/main/java/net/minecraft/server/EntityEnderman.java
++++ b/src/main/java/net/minecraft/server/EntityEnderman.java
+@@ -343,7 +343,7 @@ public class EntityEnderman extends EntityMonster {
+             BlockPosition blockposition = new BlockPosition(i, j, k);
+             IBlockData iblockdata = world.getType(blockposition);
+             IBlockData iblockdata1 = world.getType(blockposition.down());
+-            IBlockData iblockdata2 = this.a.getCarried();
++            IBlockData iblockdata2 = Block.b(this.a.getCarried(), this.a.world, blockposition); // Paper - Fix MC-124320
```
Comment (stephan-gh) [inline Spigot-Server-Patches/0348-Fix-MC-124320.patch]: Need an OBFHELPER for `Block.b` (and maybe even `this.a`)

**Review by stephan-gh — CHANGES_REQUESTED** [review]

**Review by aikar — APPROVED** [review]

**Review by stephan-gh — APPROVED** [review]

**Comment by BillyGalbreath** [comment]
~~Dont pull yet~~

**Comment by BillyGalbreath** [comment]
Ok, now it's good. (fixed the block in enderman's hand, too)

**Review by zachbr — APPROVED** [review]

**Pull request merged**